### PR TITLE
[NearbyInteraction] Bump to support Xcode 14.1

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: src/NearbyInteraction/NICompat.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using MatrixFloat4x4 = global::OpenTK.NMatrix4;
+#endif
+
+#nullable enable
+namespace NearbyInteraction {
+
+#if WATCH
+	public partial class NISession {
+
+		public virtual void SetARSession (ARSession session) => throw new InvalidOperationException (Constants.ApiRemovedGeneral);
```
Comment (dalexsoto) [inline src/NearbyInteraction/NICompat.cs:35]: I think you need the platform supported attributes probably Obsolete?

**Review by dalexsoto — CHANGES_REQUESTED** [review]

File: src/NearbyInteraction/NICompat.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/NearbyInteraction/NICompat.cs:35]: True, I forgot the obsolete. Good catch

Comment (dalexsoto) [reply src/NearbyInteraction/NICompat.cs:38]: MORE!!!

image

Comment (mandel-macaque) [reply src/NearbyInteraction/NICompat.cs:38]: Nope, this is only for watch. Watch is not present on Net, so we do not need to add anything more than the Obsolete for legacy.

File: src/NearbyInteraction/NICompat.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#endif
+
+#nullable enable
+namespace NearbyInteraction {
+
+#if WATCH
+	public partial class NISession {
+
+#if !NET
+	[Obsolete ("This method was removed and will always throw a InvalidOperationException.")]
```
Comment (mandel-macaque) [inline src/NearbyInteraction/NICompat.cs:36]:
```suggestion
		[Obsolete ("This method was removed and will always throw a InvalidOperationException.")]
```

File: src/NearbyInteraction/NICompat.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#if WATCH
+	public partial class NISession {
+
+#if !NET
+		[Obsolete ("This method was removed and will always throw a InvalidOperationException.")]
+#endif
+		public virtual void SetARSession (ARSession session) => throw new InvalidOperationException (Constants.ApiRemovedGeneral);
+
+#if !NET
+	[Obsolete ("This method was removed and will always throw a InvalidOperationException.")]
```
Comment (mandel-macaque) [inline src/NearbyInteraction/NICompat.cs:41]:
```suggestion
		[Obsolete ("This method was removed and will always throw a InvalidOperationException.")]
```

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**